signer/src/transaction_coordinator.rs

```diff
@@ -1232,13 +1233,30 @@ where
         if deposits.is_empty() && withdrawals.is_empty() {
             return Ok(None);
         }
+        // Get the amount of sBTC that can be currently minted.
+        // This is the total cap received from Emily minus the current supply.
+        let sbtc_supply = self
```

Comment:
Looks like this logic is done in several places, might be best to add a common function somewhere?

Reply:
I moved the entire logic of updating the caps in the block observer, as you suggested. That function will now both call Emily and also get the sbtc supply, if needed